pages/Documents/ConversationalAI/Third-PartyBots/CustomEndpoint/service-implementation.md

```diff
@@ -0,0 +1,371 @@
+attributes are **only** collected at the start of a conversation, and [SDES](engagement-attributes-types-of-engagement-attributes.html)
+are **only** collected if the Engagement Attributes configuration is enabled.
+This API endpoint is expected to create a conversation instance for the LivePerson
+conversation against a bot.
+
+{: .important}
+**Please Note** It is the responsibility of the Custom Endpoint service owner to
+store or update [SDES](engagement-attributes-types-of-engagement-attributes.html) information.
+
+**Endpoint**
+
+| HTTP Method | URI                                                                |
+| :---------- | :----------------------------------------------------------------- |
+| PUT         | /v1/bots/{botId}/environments/{environment}/conversations/{convId} |
+
+**Path Parameters**
+
+| Parameter   | Description                                                                                                                   |
+| :---------- | :---------------------------------------------------------------------------------------------------------------------------- |
+| botId       | Bot identifier as defined in [Vendor Configuration](third-party-bots-custom-endpoint-introduction.html#vendor-configuration)  |
+| environment | Bot environment as defined in [Vendor Configuration](third-party-bots-custom-endpoint-introduction.html#vendor-configuration) |
+| convId      | LP conversation identifier                                                                                                    |
+
+Example of the request body that is sent by Third-Party Bots can be seen
+in Figure 2.4. Please refer to [GitHub Repository](https://github.com/LivePersonInc/third-party-bots-custom-endpoint-reference-service)
+for the latest endpoint contracts/interfaces.
+
+```javascript
+{
+  "sdes": {
+    "authenticatedSdes": {
+      // ...... All SDES from Visitor Information API comes here More Info @ https://developers.liveperson.com/visit-information-api-visit-information.html
+
+    },
+    "unauthenticatedSdes": {
+      // ...... Unauthenticated SDES
+    }
+  },
+  "context": {
+    "skillId": "1234567",
+    "campaignId": 0987665546,
+    "engagementId": 123498765,
+    "type": "MESSAGING",
+    "visitor": {
+      "sharkVisitorId": "ABCDEFGHIJKLMNOPQRST",
+      "sharkSessionId": "ABcdEFGH12JKLM0iuSTU",
+      "ipAddress": "127.0.0.1",
+      "browser": "",
+      "os": "",
+      "osVersion": "",
+      "language": "en-US",
+      "consumerId": "abcdefghijklmnopqrstuvwx123456789101112131415mnopqrstuvwxyz0987654321"
+    },
+    "assignedAgentId": "abc123456-a1b2-34bc-127m-ce1a86c8602a"
+   // .... Conversation Context
+  }
+}
+```
+
+Figure 2.4 Example request body of Create Conversation endpoint
+
+#### Send Conversation Events
+
+An Event describes content sent by a brand's consumer in a conversation. This endpoint
+sends bot responses against the received consumer messages. The expected responses
```

Comment:
It is expected to respond with the actions and messages a bot wants to send as a direct response to the event